core/vm/actor/builtin/types/miner/advance_deadline_result.hpp

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "primitives/types.hpp"
+#include "vm/actor/builtin/types/miner/power_pair.hpp"
+
+namespace fc::vm::actor::builtin::types::miner {
+  using primitives::TokenAmount;
+
+  struct AdvanceDeadlineResult {
+    TokenAmount pledge_delta;
+    PowerPair power_delta;
+    PowerPair previously_faulty_power;
+    PowerPair detected_faulty_power;
+    PowerPair total_faulty_power;
+
+    inline bool operator==(const AdvanceDeadlineResult &other) const {
+      return pledge_delta == other.pledge_delta
+             && power_delta == other.power_delta
+             && previously_faulty_power == other.previously_faulty_power
+             && detected_faulty_power == other.detected_faulty_power
+             && total_faulty_power == other.total_faulty_power;
+    }
+
+    inline bool operator!=(const AdvanceDeadlineResult &other) const {
```

Comment:
also can use [`FC_OPERATOR_NOT_EQUAL(AdvanceDeadlineResult)`](https://github.com/filecoin-project/cpp-filecoin/blob/master/core/common/cmp.hpp#L12)

Reply:
I prefer to write it by myself, don't like use macros for that